### DISCUSSION TITLE
Fields ignored when populate is used

### DISCUSSION THREAD

**Discussion by yackinn** [mikro-orm/mikro-orm#6227] [Q&A]
Is there a reason why fields property in `FindOptions` is not used when `populate` is used at the same time?

```ts

const entity = await this.service.repository().findOneOrFail(
        { code: { $eq: code } },
        {
          populate,
          fields: ['kind', 'type'], // not used; all properties are on the fetched entity
        }
      )
```

Is it possible to only select certain fields from the entity and its relations?

**Comment by B4nan** [comment]
That's not how it works, you can combine those two, complete repro welcome.

**Reply by yackinn** [reply]
Thanks for your answer.
Can you point to the docs where I can find how it is supposed to be used?
Basically, I just don't want already fetched entities to be populated in child fields automatically.

```ts
// entity.ts
export class Classification {
  @OneToMany(() => Classification, (classification) => classification.parent)
  children = new Collection<Classification>(this);

  @ManyToOne(() => Classification, { nullable: true })
  parent?: Classification | null;
}

// controller.ts
const classification = await this.service.repository().findOneOrFail(
        { code: { $eq: code } },
        {
          populate: ["children"],
          fields: ['kind', 'type'],
        }
)

const shoulBeUndefined = classification.children[0].parent
```

`.parent` is populated though because I assume the main entity is managed by the entitymanager and it's in the identity map.
Is it possible to remove the properties of child properties that shouldn't be populated implicitly?

**Reply by B4nan** [reply]
> const shoulBeUndefined = classification.children[0].parent

Why do you think it should be undefined? You populate the whole `children` relation and you don't set the partial loading hint for this relation anywhere. 

Also you cant select 1:m relation without loading the m:1 part, as that is where the value actually is. 

> Is it possible to remove the properties of child properties that shouldn't be populated implicitly?

No, but you can create a new context and run the query there.

**Reply by yackinn** [reply]
```ts
const classification = await this.service.repository().findOneOrFail(
        { code: { $eq: code } },
        {
          populate: ["children"]
        }
)
```
Maybe I didn't understand the docs correctly.

`populate: ["children"]` does this mean `children` and all of it's relations are left joined into the query?
So it's basically the same as using `populate: ["children", "children.relationA", "children.relationB"]` assuming `relationA` and `relationB` are all relations of `children`.

How would I set the hint to not add relations of relations to the resulting entity object?

**Reply by B4nan** [reply]
Provide a [complete repro](https://github.com/mikro-orm/reproduction) and then I can tell, reasoning about snippets won't help much.

Also, those things should be resolved on serialization, sounds like you just check your entities on runtime instead, those will always contain cycles, you can never have a disconnected entity graph.

**Reply by B4nan** [reply]
> populate: ["children"] does this mean children and all of it's relations are left joined into the query?

No, it means only the children relation is populated. But with all the properties available, regardless of they being relations or scalars. Hard to tell if you don't like the properties being there or that they are loaded or not, but all of that should be resolved once you serialize the entity. You can never have the same entity in two places of the entity graph in two states, its always the very some object reference on runtime.

**Reply by yackinn** [reply]
Thank you for the explanation.

I understand that the object will always show referenced entities even if they're deep inside child objects.
When using `entity.toPOJO()` and then sending it to the client the parent property is still on each children entity.
So `entity.children[0].parent` is still present.

**Reply by B4nan** [reply]
> When using entity.toPOJO() 

That method ignores populate hints, its sole purpose is caching, I actually (finally) [documented this](https://mikro-orm.io/docs/serializing#caching-and-topojo) just today, you need `toObject()`.

**Reply by yackinn** [reply]
Thanks a lot. That works as intended.
